components/movementsensor/gpsrtkpmtk/gpsrtkpmtk.go

```diff
@@ -159,30 +190,20 @@ func newRTKI2C(
 		return nil, err
 	}
 
+	ntripConfig := &rtk.NtripConfig{
+		NtripURL:             newConf.NtripURL,
+		NtripUser:            newConf.NtripUser,
+		NtripPass:            newConf.NtripPass,
+		NtripMountpoint:      newConf.NtripMountpoint,
+		NtripConnectAttempts: newConf.NtripConnectAttempts,
+	}
+
```

Comment:
I am not sure what would be the best way to reconfigure the ntrip attributes. I tried adding them to to reconfig and seems like the changes don't make it back to ntrip.go. Also changing connection attempt/user/pass seems to disrupt the ntrip stream which is expected. So I am not sure if we should be adding them to reconfigure.

Reply:
if you make the entire ntripconfig an object on the rtk GPS does this behaviour hold? That's weird.

Reply:
I'd add a software test to test the reconfiguration logic so you can iterate through this faster.

Reply:
Looking into it!

Reply:
Running into issues where ntripClient is locked and reconfigure causes panic.

Reply:
Not adding ntrip attributes to reconfigure makes sense for now since ntripClient gets locked in getStream and reconfiguring causes run time error with nil pointer.

Reply:
Add the comment though, here and in rtk-serial as to why these are outside of the reconfigure function and kept in the creator.

Reply:
Added the comments on top of Reconfigure function.

Reply:
Does this mean if a user changes any of those config attributes it will only be applied when the server is restarted?

Reply:
Yes.

Reply:
we could eventually add a helper to close and start a new stream, but we should make that a follow-up ticket. Or Susmita if you'd rather explore this avenue now.

Reply:
I think @oliviamiller is right though, that's bad user experience now that I thnk about it - thanks for pointing that out.

Reply:
I'll try to add the helper functions.